### PR TITLE
Dangerous member usages analyzer rewrite

### PR DESCRIPTION
**Pull request by JeffAshton**
* Rewriting to use RegisterOperationAction
* Pulled model out of analyzer
* Added checking for passing dangerous method as a delegate

### CODE REVIEW

File: src/D2L.CodeStyle.Analyzers/ApiUsage/DangerousMemberUsages/DangerousMemberUsagesAnalyzer.cs [hunk 1]
```diff
@@ -32,172 +24,72 @@ internal sealed class DangerousMemberUsagesAnalyzer : DiagnosticAnalyzer {
 
 		private void RegisterAnalysis( CompilationStartAnalysisContext context ) {
```
Comment (JeffAshton) [inline src/D2L.CodeStyle.Analyzers/ApiUsage/DangerousMemberUsages/DangerousMemberUsagesAnalyzer.cs:25]: Merged & simplified:

image

File: src/D2L.CodeStyle.Analyzers/ApiUsage/DangerousMemberUsages/DangerousMemberUsagesAnalyzer.cs [hunk 2]
```diff
@@ -32,172 +24,72 @@ internal sealed class DangerousMemberUsagesAnalyzer : DiagnosticAnalyzer {
...
-					ctxt => {
-						if( ctxt.Node is MemberAccessExpressionSyntax propertyAccess ) {
-							AnalyzePropertyAccess( ctxt, propertyAccess, auditedAttributeType, unauditedAttributeType, dangerousProperties );
-						}
-					},
-					SyntaxKind.SimpleMemberAccessExpression
-				);
-		}
-
-		private void AnalyzeMethodInvocation(
```
Comment (JeffAshton) [inline src/D2L.CodeStyle.Analyzers/ApiUsage/DangerousMemberUsages/DangerousMemberUsagesAnalyzer.cs:75]: Simplified and moved other stuff to the model

image

**Review by j3parker — APPROVED** [review]

File: src/D2L.CodeStyle.Analyzers/Extensions/Microsoft.CodeAnalysis.cs [hunk 3]
```diff
@@ -45,7 +43,7 @@ internal static partial class RoslynExtensions {
 			return fullyQualifiedName;
 		}
 
-		public static bool IsNullOrErrorType( this ITypeSymbol symbol ) {
+		public static bool IsNullOrErrorType( [NotNullWhen( false )] this ITypeSymbol? symbol ) {
```
Comment (omsmith) [inline src/D2L.CodeStyle.Analyzers/Extensions/Microsoft.CodeAnalysis.cs:46]: Own PR, if you could

Comment (JeffAshton) [reply src/D2L.CodeStyle.Analyzers/Extensions/Microsoft.CodeAnalysis.cs:46]: > Own PR, if you could

https://github.com/Brightspace/D2L.CodeStyle/pull/799/files

**Review by omsmith — APPROVED** [review]

**Pull request merged**